tests/git.zunit

```diff
@@ -662,6 +662,299 @@
+        assert ${lines[2]} same_as diff-algorithm=minimal
+        assert ${lines[3]} same_as diff-algorithm=myers
+        assert ${lines[4]} same_as diff-algorithm=patience
+        assert ${lines[5]} same_as find-renames
+        assert ${lines[6]} same_as find-renames=
+        assert ${lines[7]} same_as ignore-all-space
+        assert ${lines[8]} same_as ignore-cr-at-eol
+        assert ${lines[9]} same_as ignore-space-at-eol
+        assert ${lines[10]} same_as ignore-space-change
+        assert ${lines[11]} same_as no-renames
+        assert ${lines[12]} same_as no-renormalize
+        assert ${lines[13]} same_as ours
+        assert ${lines[14]} same_as patience
+        assert ${lines[15]} same_as rename-threshold=
+        assert ${lines[16]} same_as renormalize
+        assert ${lines[17]} same_as subtree
+        assert ${lines[18]} same_as subtree=
+        assert ${lines[19]} same_as theirs
+    }
+
+    prefix=-qX
+    _fzf_complete_git 'git cherry-pick '
+}
+
+@test 'Testing completion: git cherry-pick --strategy-option=**' {
+    _fzf_complete() {
+        assert $# equals 4
+        assert $1 same_as '--ansi'
+        assert $2 same_as '--tiebreak=index'
+        assert $3 same_as '--'
+        assert $4 same_as 'git cherry-pick --strategy-option='
+
+        run cat
+        assert ${#lines} equals 19
+        assert ${lines[1]} same_as diff-algorithm=histogram
+        assert ${lines[2]} same_as diff-algorithm=minimal
+        assert ${lines[3]} same_as diff-algorithm=myers
+        assert ${lines[5]} same_as find-renames
+        assert ${lines[6]} same_as find-renames=
+        assert ${lines[7]} same_as ignore-all-space
+        assert ${lines[8]} same_as ignore-cr-at-eol
+        assert ${lines[9]} same_as ignore-space-at-eol
+        assert ${lines[10]} same_as ignore-space-change
+        assert ${lines[11]} same_as no-renames
+        assert ${lines[12]} same_as no-renormalize
+        assert ${lines[13]} same_as ours
+        assert ${lines[14]} same_as patience
+        assert ${lines[15]} same_as rename-threshold=
+        assert ${lines[16]} same_as renormalize
+        assert ${lines[17]} same_as subtree
+        assert ${lines[18]} same_as subtree=
+        assert ${lines[19]} same_as theirs
+    }
+
+    prefix=--strategy-option=
+    _fzf_complete_git 'git cherry-pick '
+}
+
+@test 'Testing completion: git cherry-pick --strategy-option **' {
+    _fzf_complete() {
+        assert $# equals 4
+        assert $1 same_as '--ansi'
+        assert $2 same_as '--tiebreak=index'
+        assert $3 same_as '--'
+        assert $4 same_as 'git cherry-pick --strategy-option '
+
+        run cat
+        assert ${#lines} equals 19
+        assert ${lines[1]} same_as diff-algorithm=histogram
+        assert ${lines[2]} same_as diff-algorithm=minimal
+        assert ${lines[3]} same_as diff-algorithm=myers
+        assert ${lines[4]} same_as diff-algorithm=patience
```

Comment:
I have accidentally noticed that this line is missing in `Testing completion: git pull --strategy-option=**`. Could you kindly add this there too?